content/language/word-list.md

```diff
@@ -10,8 +10,8 @@ Lists which permit or deny a set of nouns, or select enabled features.
 
 **Proposed alternative:** 
 
-* allowlist/denylist 
-* allowedNouns/deniedNouns
+* `whitelist` -> exception list, inclusion list, list of allowed X / allowed X list
+* `blacklist` -> exclusion list, prohibition list, list of unauthorized X / unauthorized X list
```

Comment:
This is less of a comment than an attempt to start a conversation to breakdown the suggestion.  Consider it partially a set of linguistic questions driven from my own mono-lingual parochialism.

It seems like we have two classes of suggestions you have made here:

1. ${qualifier} list - such as ```inclusion list``` and ```exclusion list```
2. list of ${thing} - which my brain thinks of as the [genetive case](https://en.wikipedia.org/wiki/Genitive_case) pattern

What I don't quite understand yet is why 'allow list' or 'deny list' is more difficult in some languages in the ${quantifier} list category of suggestions.  And I love to understand things :)

Reply:
One thing you will notice is that all of the things I mentioned that follow the ${quantifier} list pattern actually have direct translations and make linguistic sense. Allowlist and blocklist are completely fabricated and appear nowhere in dictionaries so raise the barrier drastically, allow list and block list are somewhat better because at least the spacing makes them easier to run through a translator, but none of those are directly translatable nor do they make sense even in English. We can only understand them because our perception of context allows us to infer - non-native speakers will struggle with such constructs.

Reply:
I appreciate you taking the time to point that out :)  It feels like we have two problems:
1. Compound words are harder to translate (neutral example intentionally using non-sense words 'foobar' vs 'foo bar')
2. All of your ${qualifier} list examples utilize *nouns* as the qualifier (even though most are formed via a -tion construction on verbs).  'allow' and 'deny' are both themselves verbs... which makes them an odd modifier to 'list'.  And 'allowtion' and 'denytion' clearly don't feel right.  I *think* I get now how the papering over of the use of verbs here would feel natural to a native speaker and wreak havoc with attempts to translate.

Let me know if this captures your thought.  And I very much appreciate your patience in helping me understand :)

Reply:
One other thing that occurs to me, that relates to one of my own personal nits about allowlist and denylist  is that I come from networking, where we use ACLs that always read something like:
```
${something to match} permit
${something to match} deny
```

which always made me want to use 'permitlist' instead of 'allowlist' (note: not advocating for that here, as that has exactly the issues you are wanting to fix, just using it for illustrative purposes).

For situations where the list is formed via:
```yaml
include:
   - A
   - D
   - E
exclude:
  - A
  - B
  - C
```
inclusion list and exclusion list seem to me to be a *perfect* match to the use case.  They pass the noun test, they relate to the verbs used in the configuration.  It matches up nicely.  I'm not quite sure what useful to do with this observation, but as you presented another occasion where it tickled my brain, I thought it worth mentioning in case smarter folks than I figure out something in this area :)

Reply:
@edwarnicke Very astute observation. Mind if I include that example in an appendix to my terminology publication and credit you? I think that quite concretely lines "exclusion list" and "inclusion list" up to be quite possibly the best replacements so far.

Points:
1. Makes actual linguistic sense in English
2. Does not rely on native-speaker inference and can be translated properly
3. Lines up with software use cases nicely

Your example goes well with point 3, and for point two, both have direct translations to other languages that can be understood by non-native speakers.

Reply:
@Nytelife26 I would be honored 😊 

You've also captured well my sense that what we have isn't a flat substitution list "replace x with y" but rather "each use of x lives in a context, and we need to find the y to replace it with which fits each context".

Reply:
@edwarnicke You were credited in the appendix of the document I linked. I also managed to draw an abstract linguistic analysis framework out from our conclusions - perhaps a good reinforcement of the one mentioned elsewhere in this project.

Reply:
@Nytelife26 Many thanks :)  I also appreciated your examples there with Spanish.  While I don't actually speak Spanish... studied enough Latin in my youth that I can follow :)

Reply:
@edwarnicke Interestingly enough, Latin is part of the reason I'm able to understand. I appreciate your understanding and positive attitude, too - the people at v8 are quite resistive at present, but I'm hoping I can work something out.

Anyways, what remains before this PR can move forward? I'm currently awaiting further review.

Reply:
@Nytelife26 I've pinged the language workstream... as they specialize in this sort of thing.  I'm just an interested amateur :)

Reply:
Jumping in from the Language workstream....First, @edwarnicke, thanks for making me aware of this conversation. You all have raised some excellent points, and I'm afraid I have to answer your questions with another question. What do translators normally do in cases where a noun or a verb is used as a modifier? That happens a lot in IT (think of a sentence like "Send the transfer packets to the network controller"). I'm not well enough versed in translation to know the answer. But it seems like an issue the translation community would have considered, and for which they might well have established a set of best practices.

Reply:
That depends largely on the engineering community present in the respective linguistic background - note that "blacklist" and "whitelist" are not exclusively engineering terms. Furthermore, it depends on the target language, given that different languages have various syntactic rules.

As for Spanish, it gets translated to "Envía los paquetes de transferencia al controlador de red" - the subject gets omitted and the sentence becomes a verbal predicate where "los paquetes de transferencia" is a direct compliment (and both "paquetes" and "transferencia" are nouns, where "transferencia" becomes a specifier for the type of packet).

It's an unusual structure, but one that can be mapped over. The requirement is that the construct must be valid in the language you're translating from.

That is a very interesting question, though, and not one I thought of before.

Reply:
Tried to read through this dialog. Very interesting observations and dialog! 

One question that comes to mind...I wonder if things like `block`/`permit` and `deny`/`allow`also serve to some extent to be a little more suggestive of *direction*?

In my mind's eye, I am thinking about something along the lines of the difference between a vector and a line-segment. One is suggestive of direction and the other isn't.

So, I wonder if `inclusion` alone can in some circumstances be more ambiguous than `permit` or `allow`?

Reply:
@Nytelife26 Question:

Would 'permitted list' 'denied list' work?  I vaguely recall that in English such past participles become adjectives... but I have no idea how that translates to other languages?

Reply:
@edwarnicke No. That implies that the list was permitted or denied, not its members, creating a distinct lack of clarity (unfortunately).

@markcmiller86 Perhaps, can you think of any examples where that is the case? I did, however, mention in my statement that a list is sometimes not needed at all. Although, I do wonder, is that out of scope? "whitelist" and "blacklist" do not mean to allow or deny - they mean to include and exclude (but often referring to a permission system).

Reply:
@Nytelife26 Ah... you are correct.  When modifying a 'list' with a noun, it implies something about the contents of the list: 'grocery list' for example, a list of (noun) groceries.   When modifying a list with an adjective (and participles like 'allowed' and 'denied' are adjectives) it would modify the 'list' itself.  For some reason I can't put my finger on my brain wants to treat adjectives applied to a list as applying to the elements of the list.

Reply:
@edwarnicke I understand honestly. It's part of how people proficient in English have adapted. We're so used to hearing people misuse the language and bend its rules without considering the consequences that we've retrained ourselves to believe the quirks are natural.

Reply:
> ... "exclusion list" and "inclusion list" up to be quite possibly the best replacements so far.

I think we have a good consensus on the alternatives here, I'm going to close this discussion for this pull request. Further discussions can come with any additional suggestions via the next PR.